src/switch/switch.test.tsx

```diff
@@ -0,0 +1,60 @@
+import React from "react";
+import {render, screen} from "@testing-library/react";
+import "@testing-library/jest-dom";
+import userEvent from "@testing-library/user-event";
+
+import {testA11y} from "../core/utils/test/testUtils";
+import Switch, {SwitchProps} from "./Switch";
+
+describe("<Switch />", () => {
+  const defaultSwitchProps: SwitchProps = {
+    isToggledOn: true,
+    onToggle: jest.fn(),
+    testid: "switch"
+  };
+
+  it("should render correctly", () => {
+    render(<Switch {...defaultSwitchProps} />);
+  });
+
+  it("should pass a11y test", async () => {
+    const {container} = render(<Switch {...defaultSwitchProps} />);
+
+    await testA11y(container, {rules: {label: {enabled: false}}});
+  });
+
+  it("should run onToggle event handler correctly", () => {
+    render(<Switch {...defaultSwitchProps} />);
+
+    userEvent.click(screen.getByRole("switch"));
+
+    expect(defaultSwitchProps.onToggle).toHaveBeenCalledTimes(1);
+  });
+
+  it("should not run onToggle event handler when isDisabled is true", () => {
+    render(<Switch {...defaultSwitchProps} isDisabled={true} />);
+
+    jest.clearAllMocks();

```

Comment:
What does this line do?

Reply:
Since I used the onToggle mock function in the previous lines, i need to clear the mock calls and instances here for consistent testing. But, i think i can move this to jest config. So it will automatically clear mock calls and instances before every test.